masonry_core/src/core/widget_pod.rs

```diff
@@ -13,89 +15,120 @@ pub struct WidgetPod<W: ?Sized> {
     inner: WidgetPodInner<W>,
 }
 
+/// A container for a widget yet to be inserted.
+#[non_exhaustive]
+#[expect(missing_docs, reason = "Names are self-explanatory.")]
+pub struct NewWidget<W: ?Sized> {
+    pub widget: Box<W>,
+    pub id: WidgetId,
+    pub options: WidgetOptions,
+    pub properties: Properties,
+}
+
+/// The options a new widget will be created with.
+#[derive(Default, Debug)]
+pub struct WidgetOptions {
+    /// The transform the widget will be created with.
+    pub transform: Affine,
+    /// The disabled state the widget will be created with.
+    pub disabled: bool,
+}
+
 // TODO - This is a simple state machine that lets users create WidgetPods
 // without immediate access to the widget arena. It's very inefficient
 // and leads to ugly code. The alternative is to force users to create WidgetPods
 // through context methods where they already have access to the arena.
 // Implementing that requires solving non-trivial design questions.
 
-pub(crate) struct CreateWidget<W: ?Sized> {
-    pub(crate) widget: Box<W>,
-    pub(crate) options: WidgetOptions,
-    pub(crate) properties: Properties,
-}
-
 enum WidgetPodInner<W: ?Sized> {
-    Create(CreateWidget<W>),
+    Create(NewWidget<W>),
     Inserted,
 }
 
-impl<W: Widget> WidgetPod<W> {
-    /// Create a new widget pod.
-    ///
-    /// In a widget hierarchy, each widget is wrapped in a `WidgetPod`
-    /// so it can participate in layout and event flow. The process of
-    /// adding a child widget to a container should call this method.
+impl<W: Widget> NewWidget<W> {
+    /// Create a new widget.
     pub fn new(inner: W) -> Self {
         Self::new_with_id(inner, WidgetId::next())
     }
 
-    /// Create a new widget pod with fixed id.
+    /// Create a new widget with fixed id.
     pub fn new_with_id(inner: W, id: WidgetId) -> Self {
-        Self::new_with_id_and_options(Box::new(inner), id, WidgetOptions::default())
+        Self {
+            widget: Box::new(inner),
+            id,
+            options: WidgetOptions::default(),
+            properties: Properties::default(),
+        }
     }
 
-    /// Create a new widget pod with properties.
+    /// Create a new widget with properties.
     pub fn new_with_props(inner: W, props: Properties) -> Self {
```

Comment:
Also probably a follow up, but having a way to have this take `Into<Properties>` and allowing creating that from various things (including a single property) would improve some of these APIs.

Reply:
Yeah, I thought about it. This is something where variadic generics would really help, though we can always use the macro tricks.